internal/pkg/scorecard/interfaces.go

```diff
@@ -130,14 +136,39 @@ func (p internalPlugin) Run() scapiv1alpha1.ScorecardOutput {
 	return res
 }
 
+// setConfigDefaults sets certain config fields to default values if they are not set
+func setConfigDefaults(config *scplugins.BasicAndOLMPluginConfig, kubeconfig string) {
+	if config.InitTimeout == 0 {
+		config.InitTimeout = 60
+	}
+	if config.ProxyImage == "" {
+		config.ProxyImage = fmt.Sprintf("quay.io/operator-framework/scorecard-proxy:%s", strings.TrimSuffix(version.Version, "+git"))
+	}
+	if config.ProxyPullPolicy == "" {
+		config.ProxyPullPolicy = v1.PullAlways
+	}
+	if config.CRDsDir == "" {
+		config.CRDsDir = scaffold.CRDsDir
+	}
+	if config.Kubeconfig == "" {
+		config.Kubeconfig = kubeconfig
+	}
+}
+
 func failedPlugin(name, desc, log string) scapiv1alpha1.ScorecardOutput {
 	return scapiv1alpha1.ScorecardOutput{
 		Results: []scapiv1alpha1.ScorecardSuiteResult{{
 			Name:        name,
 			Description: desc,
 			Error:       1,
 			Log:         log,
-		},
-		},
+			Tests: []scapiv1alpha1.ScorecardTestResult{
```

Comment:
This was done as a quick fix to make sure that we have `Error: 1` in the `SuiteResult`, since `UpdateState` just parses for the state of each test. If there are no tests, the error state gets reset to 0. I decided instead to include an empty failed test to set `Error` to 1.

This leads to another question. Should we automatically consider a suite with 0 test results as errored? That would allow us to keep the `Tests` array as nil and still have the error marked for the suite. I don't think we really need to worry about suites returning no tests (if a suite skips all of its tests, it should still report that as a test IMO), but I could be wrong.

Reply:
> since UpdateState just parses for the state of each test

Is the problem that we're resetting `Error` to zero here? https://github.com/operator-framework/operator-sdk/blob/66e78cc576efff9ec620ef2cdf35ce1a02551b88/internal/pkg/scorecard/helpers/helpers.go#L157

If so, maybe we short-circuit that function when `len(suite.Tests) == 0`, and return the suite unaltered?

Reply:
Updated